internal/auth0/organization/resource_members_test.go

```diff
@@ -0,0 +1,153 @@
+	connection_name = "Username-Password-Authentication"
+	email           = "{{.testName}}1@auth0.com"
+	password        = "MyPass123$"
+}
+
+resource "auth0_organization" "my_org" {
+	depends_on = [ auth0_user.user_1 ]
+
+	name         = "some-org-{{.testName}}"
+	display_name = "{{.testName}}"
+}
+
+resource "auth0_organization_members" "my_members" {
+	depends_on = [ auth0_organization.my_org ]
+
+	organization_id = auth0_organization.my_org.id
+	members = [ auth0_user.user_1.id ]
+}
+`
+
+const testAccOrganizationMembersWithTwoMembers = `
+resource "auth0_user" "user_1" {
+	connection_name = "Username-Password-Authentication"
+	email           = "{{.testName}}1@auth0.com"
+	password        = "MyPass123$"
+}
+
+resource "auth0_user" "user_2" {
+	depends_on = [ auth0_user.user_1 ]
+
+	connection_name = "Username-Password-Authentication"
+	email           = "{{.testName}}2@auth0.com"
+	password        = "MyPass123$"
+}
+
+resource "auth0_organization" "my_org" {
+	depends_on = [ auth0_user.user_2 ]
+
+	name         = "some-org-{{.testName}}"
+	display_name = "{{.testName}}"
+}
+
+resource "auth0_organization_members" "my_members" {
+	depends_on = [ auth0_organization.my_org ]
+
+	organization_id = auth0_organization.my_org.id
+	members = [ auth0_user.user_1.id, auth0_user.user_2.id ]
+}
+`
+
+const testAccOrganizationMembersRemoveOneMember = `
+resource "auth0_user" "user_2" {
+	connection_name = "Username-Password-Authentication"
+	email           = "{{.testName}}2@auth0.com"
+	password        = "MyPass123$"
+}
+
+resource "auth0_organization" "my_org" {
+	depends_on = [ auth0_user.user_2 ]
+
+	name         = "some-org-{{.testName}}"
+	display_name = "{{.testName}}"
+}
+
+resource "auth0_organization_members" "my_members" {
+	depends_on = [ auth0_organization.my_org ]
+
+	organization_id = auth0_organization.my_org.id
+	members = [ auth0_user.user_2.id ]
+}
+`
+
+func TestAccOrganizationMembers(t *testing.T) {
+	testName := strings.ToLower(t.Name())
+
+	acctest.Test(t, resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config:      acctest.ParseTestName(testAccOrganizationMembersPreventErasingMembersOnCreate, testName),
+				ExpectError: regexp.MustCompile("Organization with non empty members"),
+			},
+			{
+				Config: acctest.ParseTestName(testAccOrganizationMembersWithOneMember, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("auth0_organization_members.my_members", "members.#", "1"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccOrganizationMembersWithTwoMembers, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("auth0_organization_members.my_members", "members.#", "2"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccOrganizationMembersRemoveOneMember, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("auth0_organization_members.my_members", "members.#", "1"),
+				),
+			},
```

Comment:
One thing I'm noticing is that there is no test for setting a list of members to explicitly empty. Of course, users could remove the resource and delete all the members in the state, but then you're unable to track changes. While perhaps a bit of an edge case, I can see a case where a user would want this.

Not sure if this is supported in the above but if it is, would be nice to add that case. Otherwise, I would find it acceptable to address in a follow-up PR to avoid scope bloat.

Reply:
Good point 👍🏻 , added a test case in [`9d1583d` (#614)](https://github.com/auth0/terraform-provider-auth0/pull/614/commits/9d1583d60fcd55406bd65a22b4bab6e63384b907)